snap/snapcraft.yaml

```
@@ -57,33 +57,25 @@ parts:
 
   zap:
     plugin: nil
-    source: https://github.com/project-chip/zap.git
-    source-depth: 1
-    source-tag: v2023.05.04
     build-environment:
-      - NODE_VERSION: v18.16.1
+      - ZAP_VERSION: v2023.09.28
     build-packages:
       - build-essential 
-      - libpango1.0-dev 
-      - libjpeg-dev 
-      - libgif-dev 
-      - librsvg2-dev
       - wget
+      - unzip
     override-build: |
       if [[ $SNAP_ARCH == "arm64" ]]; then
-        # Install node+npm
-        NODE=node-$NODE_VERSION-linux-$SNAP_ARCH
-        wget https://nodejs.org/dist/$NODE_VERSION/$NODE.tar.xz
-        tar -xJf $NODE.tar.xz
-        export PATH=$PWD/$NODE/bin:$PATH
+        # Download and unzip the prebuilt ZAP (Zigbee Cluster Library configuration tool and generator) binary for the ARM64 architecture
+        wget https://github.com/project-chip/zap/releases/download/$ZAP_VERSION/zap-linux-$SNAP_ARCH.zip
+        unzip -o zap-linux-$SNAP_ARCH.zip
 
-        # Install zap
-        npm install
+        mkdir -p /bin && mv zap-cli /bin
```

Was the bin directory meant to be relative to the current directory, as added to the PATH?

```suggestion
        mkdir -p bin && mv zap-cli bin
```

---

Please see this [comment](https://github.com/canonical/matter-pi-gpio-commander/pull/29#issuecomment-1782714993); the zap-cli doesn't need to be moved to the bin directory.

---

It needed to be in the `./bin` directory because your environment variables pointed to that:
```
        echo "export ZAP_DEVELOPMENT_PATH=$PWD/bin" >> env
        echo "export ZAP_INSTALL_PATH=$PWD/bin" >> env
        echo "export PATH=$PWD/bin:$PATH" > env
```

---

According to the [doc](https://project-chip.github.io/connectedhomeip-doc/guides/BUILDING.html#which-zap-to-use), `ZAP_DEVELOPMENT_PATH` and `PATH` are no longer needed. Only  `ZAP_INSTALL_PATH` needs to point to where zap-linux-arm64.zip  was unpacked.

The code is as follows::
```
  wget https://github.com/project-chip/zap/releases/download/$ZAP_VERSION/zap-linux-$SNAP_ARCH.zip
  unzip -o zap-linux-$SNAP_ARCH.zip

  # Define the environment needed for the app build
  echo "export ZAP_INSTALL_PATH=$PWD" >> env
  ```